server/eventHandlers/newSessionEvents.ts

```diff
@@ -0,0 +1,111 @@
+import { IEventHandler, CWEvent, IChainEventData, SubstrateTypes } from '@commonwealth/chain-events';
+import { ValidatorInstance } from '../models/validator';
+
+export default class extends IEventHandler {
+  constructor(
+    private readonly _models
+  ) {
+    super();
+  }
+
+  /**
+    * Event handler to store new-session's validators details in DB.
+  */
+  public async handle(event: CWEvent < IChainEventData >, dbEvent) {
+    // 1) if other event type ignore and do nothing.
+    if (event.data.kind !== SubstrateTypes.EventKind.NewSession) {
+      return dbEvent;
+    }
+    const newSessionEventData = event.data;
+
+    // 2) get data from existing validators
+    const existingValidators: ValidatorInstance = await this._models.Validator.findAll({});
```

Comment:
This case was developed in the context of there will not be too many validtors in the system and most of the time current validtors will be replaced by waiting validators and vice versa with an addition of a few of the new validators at the end of the `new-session` event.

Yes, you're right, for this use case, the best possible solution which I am thinking to implement can be done in 2 simple queries:

- Directly update the Validator's `State` excluding the validators (Active/Waiting) from the `new-session` event. 
- Now get all the records for only Active/Waiting validtors which are present in the `new-session` event.

This will heavily reduce the parsing on the backend side as well for all the validators. 

would love to know your thought on this implementation.

Reply:
That sounds like a great idea! I would be sure to add comments so the flow is clear that some things happen strictly before others for a reason.

Reply:
Done and updated changes.